CMakeLists.txt

```diff
@@ -5,7 +5,7 @@
 #  Copy Policy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
 #
 
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.5)
```

Comment:
Isn't too much 3.5 as cmake minimum required?

Reply:
`CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS` requires CMake 3.4 . We can differentiate between different operating systems, but given that most of the time we are using Ubuntu Xenial with CMake 3.5 and CMake 3.5 or greater is supported (directly or using backports) in all system that we support (see https://github.com/robotology/yarp/pull/1350#issuecomment-331796555 ) I think it is not realistic to claim support for CMake < 3.5 if we are not going to test it anywhere.